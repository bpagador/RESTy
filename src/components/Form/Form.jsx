import React from 'react';
import PropTypes from 'prop-types';


const Form = ({ url, body, onSubmit, onChange }) => {
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        name="url"
        value={url}
        onChange={onChange}
        placeholder="enter api url here"
      />

      <section>
        <div>
          <label>GET
            <input type="radio" name="method" value="GET" onChange={onChange} /></label>

          <label>PUT
            <input type="radio" name="method" value="PUT" onChange={onChange} /></label>
    
          <label>POST<input type="radio" name="method" value="POST" onChange={onChange} /></label>

          <label>DELETE<input type="radio" name="method" value="DELETE" onChange={onChange} /></label>
        </div>
        
        <button>Submit</button>
      </section>
      
      <textarea
        type="text"
        name="body"
        value={body}
        onChange={onChange}

      />

      <textarea
        type="text"
        name="body"
        value={body}
        onChange={onChange}
        placeholder="results"
      />
    </form>
  );
};

Form.propTypes = {
  url: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Form;
