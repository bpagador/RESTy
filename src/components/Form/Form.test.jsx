import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('describes form jsx', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper).toMatchSnapshot();
  });
});
