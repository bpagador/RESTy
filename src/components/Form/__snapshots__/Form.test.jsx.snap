// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component describes form jsx 1`] = `
<form>
  <input
    name="url"
    placeholder="enter api url here"
    type="text"
  />
  <section>
    <div>
      <label>
        GET
        <input
          name="method"
          type="radio"
          value="GET"
        />
      </label>
      <label>
        PUT
        <input
          name="method"
          type="radio"
          value="PUT"
        />
      </label>
      <label>
        POST
        <input
          name="method"
          type="radio"
          value="POST"
        />
      </label>
      <label>
        DELETE
        <input
          name="method"
          type="radio"
          value="DELETE"
        />
      </label>
    </div>
    <button>
      Submit
    </button>
  </section>
  <textarea
    name="body"
    type="text"
  />
  <textarea
    name="body"
    placeholder="results"
    type="text"
  />
</form>
`;
